### PR TITLE
control-service: Reset termination status when job is disabled

### PR DESCRIPTION
**Pull request by doks5**
Currently, when a data job is deleted, we reset its termination status, so as to avoid issues with monitoring services reporting incorrect status. This, however, is not done in case a data job is disabled, even though the job is no longer active and its termination status should not matter.

This change modifies the DeplymentService, so that if a data job is disabled, the termination status is reset.

Testing Done: Added test.

Signed-off-by: Andon Andonov <andonova@vmware.com>

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMetricsTest.java
Comment (murphp15) [inline projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMetricsTest.java:278]: why do you care about the order?

**Comment by murphp15** [comment]
I would prefer that the test is testing the function `patchDeployment` instead of a level below that. 
Because at the moment you aren't really asserting that disabling a job clears the metrics?

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMetricsTest.java
Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMetricsTest.java:278]: Overall the whole test is miswritten and needs to be rewritten. As far as I remember some of the steps have unnecessary dependencies between them.

Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMetricsTest.java:278]: So let's keep the order for now.

**Comment by doks5** [comment]
> I would prefer that the test is testing the function `patchDeployment` instead of a level below that. Because at the moment you aren't really asserting that disabling a job clears the metrics?

Hi, @murphp15 
Could you, please, advise where the additional test should go
I searched for a test that checks if the metrics are properly cleared at job deletion, but such does not seem to exist.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMetrics.java
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMetrics.java:215]: You should not name a method based on who is using it but based on what it does. We may want to clear the same gauges on deploy Remove as well. 

Which is a follow-up question. Don't we ?

**Comment by tozka** [comment]
Looks in the right direction. I think we need to see what happens when we remove a job deployment as well (I honestly do not know and haven't time to check the code so it's up to you). I am also fine if you decide to handle this case in subsequent PR. 

Don't wait for my approval since I am going on vacation :)

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMetrics.java
Comment (doks5) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/monitoring/DataJobMetrics.java]: Agree, the method name didn't make sense. Changed, thanks

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/deploy/DeploymentServiceTest.java
Comment (doks5) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/deploy/DeploymentServiceTest.java]: Renamed, thanks

**Review by murphp15 — APPROVED** [review]

**Pull request merged**
